pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -182,10 +184,17 @@ public SegmentDataManager acquireSegment(@Nonnull String segmentName) {
     if (segmentDataManager != null && segmentDataManager.increaseReferenceCount()) {
       return segmentDataManager;
     } else {
+      handleMissingSegment(segmentName);
       return null;
     }
   }
 
+  private void handleMissingSegment(String segmentName) {
+    // could not find segment
+    LOGGER.error("Could not find segment " + segmentName + " for table " + _tableNameWithType);
```

Comment:
This will be triggered whenever new segment is added.
In `SegmentFetcherAndLoader.addOrReplaceOfflineSegment(...)`, we always get the segment metadata. which will always call `acquireSegment(...)`.

Reply:
@chenboat I guess this is why we are getting many error logs ^^ I think it's more like a false alarm.

Reply:
@sunithabeeram @Jackie-Jiang ^^